### PR TITLE
Start documentation of common test failures

### PR DESCRIPTION
**Pull request by cecille**
Per a request from an ATL, this is the start of an ongoing documentation project to help ATLs and companies root cause common test failure problems that are real failures.

### CODE REVIEW

File: docs/common_test_failures/TC-DA-1.7.md
Comment (akrithika27) [inline docs/common_test_failures/TC-DA-1.7.md:30]: Remove "\" at the end of the line.

Comment (cecille) [inline docs/common_test_failures/TC-DA-1.7.md:30]:
```suggestion
   <td>The two devices under test use the same DAC public key (aka, they are the same DAC). This is disallowed under section 6.2.2, which states “All commissionable Matter Nodes SHALL include a Device Attestation Certificate (DAC) and corresponding private key, unique to that Device.
```

Comment (cecille) [reply docs/common_test_failures/TC-DA-1.7.md]: done.

**Comment by ReneJosefsen** [comment]
I fully support this effort, but the way I interpreted the request raised in the TT meeting, was that the scripts them self was more verbose in what failed and as you pointed out the example provided was a script where the assert message was not super clear.

E.g. when we execute a test in the YAML framework there is a common message if the assert fails, at this moment I think it is a bit the "wild west" for the python scripts since most of them use the assert functionality directly. So maybe we should define a common text for these asserts as well as expose which step (if applicable/available) the failure occurred in?

**Review by ReneJosefsen — APPROVED** [review]

**Comment by cecille** [comment]
> I fully support this effort, but the way I interpreted the request raised in the TT meeting, was that the scripts them self was more verbose in what failed and as you pointed out the example provided was a script where the assert message was not super clear.
> 
> E.g. when we execute a test in the YAML framework there is a common message if the assert fails, at this moment I think it is a bit the "wild west" for the python scripts since most of them use the assert functionality directly. So maybe we should define a common text for these asserts as well as expose which step (if applicable/available) the failure occurred in?

yeah, I think that's totally possible. I have a PR up to make some changes to the basic framework to start allowing callbacks to the TH to mark steps as passed, failed or skipped. The TH team is still trying to figure out exactly what they want to do there, but once that lands, the info will be available. There's also a separate mechanism used by the device basic composition tests where we mark a problem using a specific problem notice format and continue with the test until it's explicitly failed. That mechanism is interesting for some of the whole-device tests, where we don't want to keep stepping through issues one by one, but see them all in their entirety. That format is a bit different, and it's not so easily tied to a specific step, but more to a specific problem path. Still, IMO, those are very parseable and the device basic composition test steps don't really help anway. 

Either way, waiting on the TH team for their feedback, but this should be fairly easily covered once the TH integration starts.

**Review by tehampson — APPROVED** [review]

File: docs/common_test_failures/README.md
Comment (rbultman) [inline docs/common_test_failures/README.md:3]:
```suggestion
The documents in this folder are intended to be used to help root-cause common test failures,
especially in cases where the underlying cause of the failure may not be immediately obvious from the test step or expected outcomes.
```

**Review by rbultman — APPROVED** [review]

**Review by jmeg-sfy — APPROVED** [review]

**Comment by jmeg-sfy** [comment]
> I fully support this effort, but the way I interpreted the request raised in the TT meeting, was that the scripts them self was more verbose in what failed and as you pointed out the example provided was a script where the assert message was not super clear.
> 
> E.g. when we execute a test in the YAML framework there is a common message if the assert fails, at this moment I think it is a bit the "wild west" for the python scripts since most of them use the assert functionality directly. So maybe we should define a common text for these asserts as well as expose which step (if applicable/available) the failure occurred in?

yes indeed , best is to have good error feedback (and verbose, not too verbose though)

**Review by akrithika27 — APPROVED** [review]

**Review by raju-apple — APPROVED** [review]

**Review by antonio-amjr — APPROVED** [review]

**Pull request merged**
